src/bin/cargo-miri.rs

```diff
@@ -347,6 +346,10 @@ path = "lib.rs"
         command.env("RUSTC", find_miri());
     }
     command.env("MIRI_BE_RUSTC", "1");
+    // Make sure there are no other wrappers or flags getting in our way
+    // (Cc https://github.com/rust-lang/miri/issues/1421).
+    command.env_remove("RUSTC_WRAPPER");
```

Comment:
Maybe it shouldn't be done in invisible to a user, manner? I.e. throw an error if there are certain variables that conflict with miri?

Reply:
Hm, good point. Throwing an error might be too extreme though. What about printing a warning, similar to what we do on `MIRI_SYSROOT=something cargo miri setup`?

Reply:
well, before I faced some errors in `miri` setup with the caching I didn't even know it removes certain variables. It would be at least nice to add the warnings that "x and y are being overriden".

Reply:
Yes we have such a warning for `RUSTC_WRAPPER` now.

Reply:
Yeah, I've noticed it right after this comment. Thanks a bunch for this!